lib/lintPostcssResult.js

```diff
@@ -131,4 +135,14 @@ function isFixCompatible({ stylelint }) {
 	return true;
 }
 
-module.exports = lintPostcssResult;
+/**
+ * @param {PostcssResult} result
+ * @param {string} ruleName
+ * @returns {void}
+ */
+function warnDeprecatedRule(result, ruleName) {
+	result.warn(
+		`The "${ruleName}" rule is deprecated and will be removed in the next major release.`,
```

Comment:
```suggestion
		`The "${ruleName}" rule is deprecated.`,
```

Let's keep it succinct in the terminal to avoid a wall of text.

Reply:
What about having just one sentence with a list instead of multiple sentences?
e.g. `The "string-quotes", "unit-case" and "foo-bar" rules are deprecated and will be removed in the next major release.`

Reply:
Thanks for the advice. I've simplified warning messages via ff83ba86c. Here is a screenshot:

<img width="938" alt="image" src="https://user-images.githubusercontent.com/473530/212265406-ff6d9230-6b2f-4a43-8bd5-628e6fe7c98d.png">

I keep multiple sentences because each message may have a `reference`, and it's hard to include reference(s) in one sentence. See the code below:

https://github.com/stylelint/stylelint/blob/ff83ba86caa9d5b3f9db47903fbb29ec1f6bb98d/lib/formatters/stringFormatter.js#L59-L61

Instead, I've avoided repeated "Deprecation Warning" messages in each line and appended a " - " prefix to each line.

What about this change?

Reply:
Let's take stylelint config standard as the baseline.
Let's say a user resets about 19% of the deprecated rules; that clocks at around 12 lines out of potentially 64.
In that scénario I would prefer a counter and the usage of `…` or `etc.` for more than 3.
e.g.
```js
`${count} deprecation warnings for rules ${rules[0]},  ${rules[1]}, ${rules[2]}, etc.`
```

Reply:
> What about this change?

SGTM, thank you.

Reply:
@Mouvedia I understand your concern. But, if `reference` is specified with a rule deprecation, how should we display it with `…` or `etc.`?　I believe we need to consider also plugin rules, not only the built-in ones.

Reply:
Alright, as long as we a ENV variable—or something similar—that hides those warnings it should be OK.
@ybiquitous Does the `--quiet` option hide these warnings?

Reply:
> Does the `--quiet` option hide these warnings?

I'm afraid the answer is **no** because these warnings are not related to a *warning* severity produced by lint rules. We currently don't have a way to suppress this kind of warning, except for using a custom formatter.